### PR TITLE
Study/Service descriptions support markdown

### PR DESCRIPTION
**Pull request by odeimaiz**
Took it over from @ignapas 

## What do these changes do?
- [x] Second version of the study details view @ignapas 
- [x] StudyBrowser bugs corrected @ignapas 
- [x] Some code fixes for the Markdown class @ignapas 
- [x] Marked.js modified to fit markdown images to width 100% 
- [x] Add thumbnail and badges to service schema
- [x] Study description supports markdown
- [x] Service description supports markdown

## Related issue number
#1012 

## Screenshots
Study:
image

Service:
SleeperDetails

## How to test
Play with selection, the buttons, and editing the data. Try adding markdown to the description.
Suggestions welcomed.

### CODE REVIEW

File: services/web/client/source/class/qxapp/component/metadata/StudyDetails.js [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+ * Copyright: 2019 IT'IS Foundation - https://itis.swiss
+ * License: MIT - https://opensource.org/licenses/MIT
+ * Authors: Ignacio Pascual (ignapas)
+ *          Odei Maiz (odeimaiz)
+ */
+
+qx.Class.define("qxapp.component.metadata.StudyDetails", {
+  extend: qx.ui.core.Widget,
+
+  construct: function(study, isTemplate) {
```
Comment (oetiker) [inline services/web/client/source/class/qxapp/component/metadata/StudyDetails.js]: there seems to be a lack of documentation

File: api/specs/shared/schemas/node-meta-v0.0.1-converted.yaml [hunk 2]
```diff
@@ -41,6 +41,43 @@ properties:
...
     example:
       - Fast Counter
+  thumbnail:
+    type: string
+    description: 'url to the thumbanil'
+    example:
+      - https://user-images.githubusercontent.com/32800795/61083844-ff48fb00-a42c-11e9-8e63-fa2d709c8baf.png
+  badges:
+    type: array
+    minItems: 0
```
[the remark this answers is not in the document]
Comment (odeimaiz) [reply api/specs/shared/schemas/node-meta-v0.0.1-converted.yaml]: Correct
https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.3.3

**Comment by odeimaiz** [comment]
> This is very nice but there are still some more inconsistencies between the different UI sections, e.g. why i can search services and not studies? I can also create studies but not services (e.g. creation of services might open a website explaining why to submit them etc ). I think the three tabs in the UI should have similar functionality. At the end of the day, each tab represents a resource (studies, services, data). I want to list (w/ filter), create/delete, display (e.g. info, description, meta), edit ...

This sounds very good and it would make the UX very smooth, but we are far from having CRUD operations for all three type of resources. I will create a Backlog issue.

File: api/specs/shared/schemas/node-meta-v0.0.1-converted.yaml [hunk 3]
```diff
@@ -41,6 +41,42 @@ properties:
     description: 'short, human readable name for the node'
     example:
       - Fast Counter
+  thumbnail:
```
Comment (sanderegg) [inline api/specs/shared/schemas/node-meta-v0.0.1-converted.yaml]: did you generate the file using the converter script?

File: services/web/client/source/class/qxapp/component/metadata/StudyDetails.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+ *    this.add(studyDetails);
+ * </pre>
+ */
+
+qx.Class.define("qxapp.component.metadata.StudyDetails", {
+  extend: qx.ui.core.Widget,
+
+  /**
+    * @param study {Object|qxapp.data.model.Study} Study (metadata)
+    * @param maxHeight {Integer} Max Height of the thumbnail
```
Comment (ignapas) [inline services/web/client/source/class/qxapp/component/metadata/StudyDetails.js]: Is it necessary to pass this to the widget?

**Pull request merged**